### PR TITLE
[WIP] fix: better theme name consistency.

### PR DESCRIPTION
**Pull request by disordinary**
Closes: https://github.com/TryGhost/Ghost/issues/7313
- Adds `getFileName` function to storage/base.js which replaces non `A-Z0-9@.` chacracters with `-`
- modifies /api/theme.js so that `zip.shortName` is consistent with the filename rules as above.

### CODE REVIEW

**Comment by kirrg001** [comment]
Thanks @disordinary ! Will not review before Wednesday next week, just that you know :)

File: core/test/unit/storage/index_spec.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ describe('storage: index_spec', function () {
             (chosenStorage instanceof localFileStorage).should.eql(true);
         });
 
+        it('escape non accepted characters in filenames', function () {
```
Comment (kirrg001) [inline core/test/unit/storage/index_spec.js]: could you please move this test into a test file named `storage/base_spec.js`?
the test you have added is now part of a test section which covers loading storage adapters.

and if you have time, the logic you have added is not covered by a test. you could copy this test https://github.com/TryGhost/Ghost/blob/master/core/test/functional/routes/api/themes_spec.js#L73 and just upload a different theme with some special characters.
thanks 👍

File: core/server/api/themes.js [hunk 2]
```diff
@@ -54,19 +54,19 @@ themes = {
                 );
             })
             .then(function () {
-                return storageAdapter.exists(config.paths.themePath + '/' + zip.shortName);
+                return storageAdapter.exists(config.paths.themePath + '/' + storageAdapter.getFileName(zip.shortName));
```
Comment (kirrg001) [inline core/server/api/themes.js]: readability suggestion: 
on top do this:
```
zip.shortName = storageAdapter.getFileName(zip.shortName);
```

**Comment by ErisDS** [comment]
This PR is adding a new method to the base storage adapter, which results in a breaking change for all storage adapters. Not sure if this has been realised as it is being used throughout the code as though it is a required function, but hasn't been added to the required functions list.

If there is another way we can achieve the same result without making a breaking change, that would be preferable, as we've only just made one.

If not, then this will need to be flagged up: clearly noted in the PR and on the issue. The wiki documentation will need to be updated and customarily we post an issue on all of the storage adapters linked in the wiki as well. 

TLDR; a breaking change is a lot more work than a non-breaking change.

**Comment by disordinary** [comment]
I've renamed getFileName to be clearer about what it does.

**Comment by ErisDS** [comment]
Cool, I've paused the build as we're short of travis cycles today. Are you ok to get the other changes made, squash and get this ready to go? I think it has been fully reviewed now :)

**Comment by disordinary** [comment]
Yes, it should have addressed each point raised but for some reason the old changes have merged back in. I'll send a PR through shortly.

**Comment by ErisDS** [comment]
rebased & pushed to #7380
